cmd/boostd/backup.go

```diff
@@ -0,0 +1,274 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path"
+	"time"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/urfave/cli/v2"
+	"gopkg.in/cheggaaa/pb.v1"
+
+	"github.com/filecoin-project/boost/node"
+	"github.com/filecoin-project/lotus/chain/store"
+	"github.com/filecoin-project/lotus/lib/backupds"
+	lotus_repo "github.com/filecoin-project/lotus/node/repo"
+	dstore "github.com/ipfs/go-datastore"
+)
+
+var fm = map[string]fs.FileMode{"api": 0644, "boost.db": 0644, "boost.logs.db": 0644, "config.toml": 0664, "storage.json": 0644, "token": 0600}
+
+var backupCmd = &cli.Command{
+	Name:  "backup",
+	Usage: "Perorms offline backup of the Boost subsystem",
+	Flags: []cli.Flag{
+		&cli.PathFlag{
+			Name:     "path",
+			Usage:    "Specify a directory to create backup. Must be inside home directory.",
+			Required: false,
+		},
+	},
+	Before: before,
+	Action: func(cctx *cli.Context) error {
+
+		boostRepoPath := cctx.String(FlagBoostRepo)
+
+		r, err := lotus_repo.NewFS(boostRepoPath)
+		if err != nil {
+			return err
+		}
+		ok, err := r.Exists()
+		if err != nil {
+			return err
+		}
+		if !ok {
+			return fmt.Errorf("repo at '%s' is not initialized", cctx.String(FlagBoostRepo))
+		}
+
+		lr, err := r.LockRO(node.Boost)
+		if err != nil {
+			return fmt.Errorf("locking repo: %w", err)
+		}
+		defer lr.Close()
+
+		mds, err := lr.Datastore(context.TODO(), "/metadata")
+		if err != nil {
+			return fmt.Errorf("getting metadata datastore: %w", err)
+		}
+
+		bds, err := backupds.Wrap(mds, backupds.NoLogdir)
+		if err != nil {
+			return err
+		}
+
+		bpath, err := homedir.Expand(cctx.Path("path"))
+		if err != nil {
+			return fmt.Errorf("expanding backup directory path: %w", err)
+		}
+
+		bkp_dir := path.Join(bpath, "boost_backup_"+time.Now().Format("20060102150405"))
```

Comment:
In our code we use camelCase rather than snake_case. Let's change all the variables to be camel case.
```suggestion
		bkpDir := path.Join(bpath, "boost_backup_"+time.Now().Format("20060102150405"))
```